qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/TypeRemovalIT.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.upgrades;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.util.EntityUtils;
+import org.opensearch.Version;
+import org.opensearch.client.Request;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.Response;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static org.opensearch.rest.action.search.RestSearchAction.TOTAL_HITS_AS_INT_PARAM;
+
+public class TypeRemovalIT extends AbstractRollingTestCase {
+
+    public void testTypeRemovalIndexing() throws IOException {
+        final String indexName = "test_index_with_mapping";
+        final String indexWithoutTypeName = "test_index_without_mapping";
+        final String templateIndexName = "template_test_index";
+        final String templateName = "test_template";
+        final String indexNamePattern = "template_test*";
+
+        final String mapping = "\"properties\":{\"f1\":{\"type\":\"keyword\"},\"f2\":{\"type\":\"keyword\"}}";
+        final String typeMapping = "\"_doc\":{" + mapping +"}";
+
+        switch (CLUSTER_TYPE) {
+            case OLD:
+                if (UPGRADE_FROM_VERSION.before(Version.V_2_0_0)) {
```

Comment:
Against 3.0, it will run against the previous wire compatible versions (v2.x.x). And yes, it will be a NoOp in the versions ahead.
Let me branch it off 2.0 instead.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/2901

Reply:
Looks like we also need this test in `2.x` branch? 😂